### ISSUE TITLE
HoloMap crash with matplotlib using %opts Points (s=...)

### ISSUE THREAD

**Issue by alexey-petrenko** [pyviz/holoviews#3439]
Dear Holoviews developers,

There seems to be a problem with hv.Points plot in HoloMap in the recent holoviews (v 1.11.0). If I set %opts Points (s=50) or any other value it crashes:

I'm trying to reproduce the [holomap example](http://holoviews.org/reference/containers/matplotlib/HoloMap.html) using hv.Points instead of hv.Curve:
```
import numpy as np
import holoviews as hv
hv.extension('matplotlib')

%opts Points (s=50)

frequencies = [0.5, 0.75, 1.0, 1.25]

def sine_curve(phase, freq):
    xvals = [0.1* i for i in range(100)]
    #return hv.Curve((xvals, [np.sin(phase+freq*x) for x in xvals]))
    return hv.Points((xvals, [np.sin(phase+freq*x) for x in xvals]))

curve_dict = {f:sine_curve(0,f) for f in frequencies}

hv.HoloMap(curve_dict, kdims='frequency')
```

The error message:

```
...
C:\ProgramData\Anaconda3\lib\site-packages\holoviews\plotting\mpl\chart.py in update_handles(self, key, axis, element, ranges, style)
    668         paths.set_offsets(np.column_stack([xs, ys]))
    669         if 's' in style:
--> 670             paths.set_sizes(style['s'])
    671         if 'vmin' in style:
    672             paths.set_clim((style['vmin'], style['vmax']))

C:\ProgramData\Anaconda3\lib\site-packages\matplotlib\collections.py in set_sizes(self, sizes, dpi)
    871         else:
    872             self._sizes = np.asarray(sizes)
--> 873             self._transforms = np.zeros((len(self._sizes), 3, 3))
    874             scale = np.sqrt(self._sizes) * dpi / 72.0 * self._factor
    875             self._transforms[:, 0, 0] = scale

TypeError: len() of unsized object
```

I was wondering if you know how to change the point size correctly in this case?

In the earlier holoviews version 1.9.1 the same example works perfectly well.

Many thanks!
Best regards,
/Alexey

**Comment by alexey-petrenko** [comment]
I'm sorry, you have closed this issue, but it is still present in the latest holoviews (v 1.11.2):

The same example gives the same TypeError:

I was wondering if there is some workaround to avoid this error? I would really like to change the default size of Points (they are too big for large data sets).

`import numpy as np
import holoviews as hv

print(hv.__version__)

hv.extension('matplotlib')

%opts Points (s=50)

frequencies = [0.5, 0.75, 1.0, 1.25]

def sine_curve(phase, freq):
    xvals = [0.1* i for i in range(100)]
    #return hv.Curve((xvals, [np.sin(phase+freq*x) for x in xvals]))
    return hv.Points((xvals, [np.sin(phase+freq*x) for x in xvals]))

curve_dict = {f:sine_curve(0,f) for f in frequencies}

hv.HoloMap(curve_dict, kdims='frequency')`

> 

1.11.2

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
C:\ProgramData\Anaconda3\lib\site-packages\IPython\core\formatters.py in __call__(self, obj, include, exclude)
    968 
    969             if method is not None:
--> 970                 return method(include=include, exclude=exclude)
    971             return None
    972         else:

...

C:\ProgramData\Anaconda3\lib\site-packages\holoviews\plotting\mpl\chart.py in update_handles(self, key, axis, element, ranges, style)
    671         paths.set_offsets(np.column_stack([xs, ys]))
    672         if 's' in style:
--> 673             paths.set_sizes(style['s'])
    674         if 'vmin' in style:
    675             paths.set_clim((style['vmin'], style['vmax']))

C:\ProgramData\Anaconda3\lib\site-packages\matplotlib\collections.py in set_sizes(self, sizes, dpi)
    871         else:
    872             self._sizes = np.asarray(sizes)
--> 873             self._transforms = np.zeros((len(self._sizes), 3, 3))
    874             scale = np.sqrt(self._sizes) * dpi / 72.0 * self._factor
    875             self._transforms[:, 0, 0] = scale

TypeError: len() of unsized object


Many thanks!
/Alexey

**Comment by philippjfr** [comment]
Can't seem to reproduce this issue now, could you also report your matplotlib version?

**Comment by philippjfr** [comment]
Nevermind the fix didn't actually make it into 1.11.2. I'll be releasing 1.11.3 shortly so just update to that.
